src/main.rs

```diff
@@ -55,6 +55,36 @@ use terminal::Terminal;
 
 type Result<T> = ::std::result::Result<T, Error>;
 
+#[derive(Clone)]
+enum FileSet {
+    Extensions(Vec<String>),
+    Glob {
+        matches: Vec<String>,
+        case_insensitive: bool,
+    },
+}
+
+fn get_file_set(matches: &clap::ArgMatches) -> Option<FileSet> {
+    if matches.is_present("extensions") {
```

Comment:
nit: this could be written `if let Some(files) = matches.values_of_lossy("extensions")` (and similarly for the other cases) to avoid the `is_present` checks